### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by dmbichko**
**_**This PR contains the code to complete all the necessary requirements of task 2**_**
Once the pipeline is complete, I will upload all the necessary screenshots to the screenshot folder. 

task2

Evaluation Criteria (100/100 points for covering all criteria) ✔️ Success

Terraform Code Implementation (50/50 points) ✔️ Success
Terraform code is created to configure the following:
VPC
2 public subnets in different AZs
2 private subnets in different AZs
Internet Gateway
Routing configuration:
Instances in all subnets can reach each other
Instances in public subnets can reach addresses outside VPC and vice-versa

Code Organization (10/10 points) - ✔️ Success
Variables are defined in a separate variables file.
Resources are separated into different files for better organization.
Verification (10 points)

Terraform plan is executed successfully. ✔️ Success
A resource map screenshot is provided (VPC -> Your VPCs -> your_VPC_name -> Resource map).

Additional Tasks (30/30 points) ✔️ Success

Security Groups and Network ACLs (5/5 points) ✔️ Success
Implement security groups and network ACLs for the VPC and subnets.

Bastion Host (5/5 points) ✔️ Success
Create a bastion host for secure access to the private subnets.

NAT is implemented for private subnets (10 points) ✔️ Success
Orginize NAT for private subnets with simpler or cheaper way
Instances in private subnets should be able to reach addresses outside VPC

Documentation (5 points) ✔️ Success
Document the infrastructure setup and usage in a README file.

Submission (5 points) ✔️ Success
A GitHub Actions (GHA) pipeline is set up for the Terraform code.

### CODE REVIEW

File: .github/workflows/terraform_deploy.yml [hunk 1]
```diff
@@ -87,7 +91,7 @@ jobs:
           aws-region: ${{ env.AWS_REGION }}
 
       - name: Terraform Plan
-        run: terraform plan -input=false -var="aws_account_id=${{ env.AWS_ACCOUNT_ID }}"
+        run: terraform plan -input=false -var="aws_account_id=${{ env.AWS_ACCOUNT_ID }}" -var="bastion-ssh-key=${{ env.BASTION_SSH_KEY }}" -var="ec2-ssh-key=${{ env.EC2_SSH_KEY }}"
```
Comment (Romicusblr) [inline .github/workflows/terraform_deploy.yml:94]: imho using TF_VAR_account_id more elegant solution

File: output.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Elastic IP for Bastion
+resource "aws_eip" "bastion" {
```
Comment (Romicusblr) [inline output.tf:2]: strange resourse block in outputs

File: vpc.tf [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+# Private Route Table Association
+resource "aws_route_table_association" "private_routers" {
+  count          = length(aws_subnet.private_subnets[*].id)
+  route_table_id = aws_route_table.private_rt[count.index].id
+  subnet_id      = element(aws_subnet.private_subnets[*].id, count.index)
+}
+
+
+# Outputs
```
Comment (Romicusblr) [inline vpc.tf:117]: imho outputs should be in separate file

File: output.tf (hunk 2 above)
Comment (dmbichko) [reply output.tf:2]: Ok. I've done it another way.

File: vpc.tf (hunk 3 above)
Comment (dmbichko) [reply vpc.tf:117]: You're right. Moved to a appropriate file.

**Pull request merged**